### PR TITLE
rename public variable for Fortran codegen string

### PR DESCRIPTION
**Pull request by gxyd**
change `s` to `fortran_code`, as that's self explanatory

### CODE REVIEW

**Review by certik — APPROVED** [review]
I think that's fine. @Thirumalai-Shaktivel what do you think?

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 1]
```diff
@@ -28,7 +28,7 @@ enum Precedence {
 class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
 {
 public:
-    std::string s;
+    std::string fortran_code;
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_fortran.cpp:31]: Let's keep it the same as the other backend.
In C, CPP, and Julia, we use `src`, so let's use the same here as well?

Or 

Let's keep it simple, something like `code`, `out`, ...

Comment (certik) [reply src/libasr/codegen/asr_to_fortran.cpp:31]: Or we can just leave `s`. :)

Comment (gxyd) [reply src/libasr/codegen/asr_to_fortran.cpp:31]: when I looked at the class functions directly, I didn't know what `s` is supposed to be, so I went to it's definition, and without comments here, it took me a minute, by observing its use in other places to be sure for what it's purpose is. `src` is more self-explanatory than `s` is.

Comment (certik) [reply src/libasr/codegen/asr_to_fortran.cpp:31]: Ok, let's use `src`.

Comment (gxyd) [reply src/libasr/codegen/asr_to_fortran.cpp]: Done, I've renamed it to `src` and put a comment as well.

**Review by Thirumalai-Shaktivel — APPROVED** [review]
LGTM, thank you!

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 2]
```diff
@@ -28,7 +28,11 @@ enum Precedence {
 class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
 {
 public:
-    std::string s;
+    // `src` acts as a buffer that accumulates the generated Fortran source code
+    // as the visitor traverses all the ASR nodes of a program. Each visitation method
+    // appends the Fortran translation of the current node to `src`, building up
+    // the complete source code incrementally
```
Comment (certik) [inline src/libasr/codegen/asr_to_fortran.cpp:34]:
```suggestion
    // `src` acts as a buffer that accumulates the generated Fortran source code
    // as the visitor traverses all the ASR nodes of a program. Each visitor method
    // uses `src` to return the result, and the caller visitor uses `src` as the return value of the callee visitors it calls. 
    // The Fortran complete source code is then recursively constructed using `src`.
```

File: src/libasr/codegen/asr_to_fortran.cpp [hunk 3]
```diff
@@ -1620,53 +1625,71 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
...
+        r += ", ";
+        visit_expr(*x.m_mask);
+        r += src;
+        if (x.m_vector) {
+            r += ", ";
+            visit_expr(*x.m_vector);
+            r += src;
+        }
+        r += ")";
+        src = r;
```
Comment (certik) [inline src/libasr/codegen/asr_to_fortran.cpp:1656]: I think I resolved the conflict incorrectly. This function was removed in 4065ff820e824634c40cc44fb655b8e2d7243861.

**Comment by gxyd** [comment]
I've force pushed the changes onto `main` now, should probably be good now.

**Review by certik — APPROVED** [review]
I think this is good now. Thanks!

**Pull request merged**